sources/en/f/freewebnovel.py

```diff
@@ -30,17 +29,17 @@ def parse_search_item(self, tag: Tag) -> SearchResult:
 
     def parse_title(self, soup: BeautifulSoup) -> None:
         tag = soup.select_one(".m-desc h1.tit")
-        if not isinstance(tag, Tag):
-            raise LNException("No title found")
+        assert isinstance(tag, Tag), "No title found"
```

Comment:
as far as I remember, assert only works when debagging?

Reply:
it just raises an exception with the second parameter if first once is falsy. doesn't matter if you are debugging or not

Reply:
@dipu-bd 
![image](https://user-images.githubusercontent.com/11484976/193681148-ad5e38a5-1e58-4192-bbdb-f39befc1cf99.png)

Reply:
> In the current implementation, the built-in variable [__debug__](https://docs.python.org/3.6/library/constants.html#__debug__) is True under normal circumstances, False when optimization is requested (command line option [-O](https://docs.python.org/3.6/using/cmdline.html#cmdoption-o)). The current code generator emits no code for an assert statement when optimization is requested at compile time.

I did not know that! According to their docs, python runs as `__debug__` by default, and I never tried with `-O` or `-OO`

Reply:
Python is a bit more complicated, but I've written in many other languages. For example, in C# there is a debug and release version of the program. Assert works in the debug, but it is not compiled in the release. Also in many other languages.

I thought you were a God of Python and you know everything. :joy_cat:

Reply:
> I thought you were a God of Python and you know everything.

Ha ha ha! 

I am kind curious to know how you came to think that! I think that most people see this project and curse me silently because of such messy beginner level code.